### PR TITLE
(SIMP-3283) user guide auth.conf updates

### PR DESCRIPTION
**Pull request by nick-markowski**
- Updated the user guide with instructions to back up legacy auth.conf
  and modify new auth.conf, need be.

SIMP-3283 #comment done in docs

### CODE REVIEW

File: docs/user_guide/HOWTO/Upgrade_SIMP.rst [hunk 1]
```diff
@@ -99,3 +99,23 @@ of your new Puppet server into the existing environment.
 
    You should *always* start migration with a small number of
    **least critical** clients!
+
+Significant Updates
+-------------------
+
+Upgrading From SIMP-6.0.0-0 to SIMP-6.0.0++
```
Comment (jeefberkey) [inline docs/user_guide/HOWTO/Upgrade_SIMP.rst]: You should be more verbose about what `++` means

File: docs/user_guide/HOWTO/Upgrade_SIMP.rst [hunk 2]
```diff
@@ -99,3 +99,23 @@ of your new Puppet server into the existing environment.
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Puppetserver auth.conf
+""""""""""""""""""""""
+
+Legacy auth.conf has been deprecated.  The puppetserver's auth.conf is now
+managed by Puppet.
+
+1. Back up legacy puppet auth.conf ``<puppet confdir>/auth.conf``, before
```
Comment (jeefberkey) [inline docs/user_guide/HOWTO/Upgrade_SIMP.rst]: This has a real location that should be the same in SIMP 6 installs.

Comment (nick-markowski) [inline docs/user_guide/HOWTO/Upgrade_SIMP.rst]: Not sure what you're trying to say here. Auth.conf moved. Do you want me to fullpath the confdir?

Comment (jeefberkey) [inline docs/user_guide/HOWTO/Upgrade_SIMP.rst]: Yeah

File: docs/user_guide/HOWTO/Upgrade_SIMP.rst [hunk 3]
```diff
@@ -99,3 +99,23 @@ of your new Puppet server into the existing environment.
...
+Upgrading From SIMP-6.0.0-0 to A Later Version
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Puppetserver auth.conf
+""""""""""""""""""""""
+
+Legacy auth.conf has been deprecated.  The puppetserver's auth.conf is now
+managed by Puppet.
+
+1. Back up legacy puppet auth.conf ``/etc/puppetlabs/puppet/auth.conf``, before
```
Comment (trevor-vaughan) [inline docs/user_guide/HOWTO/Upgrade_SIMP.rst]: You don't actually have to do this. Our module will back it up and move it out of the way for you.

Comment (nick-markowski) [inline docs/user_guide/HOWTO/Upgrade_SIMP.rst]: We can remove it

**Pull request merged**